### PR TITLE
feat(VBtn): add new styling property plain

### LINKED ISSUE #3026: [Feature Request] New button variant: plain (no background hover)
### New Functionality
For some UI I prefer buttons with no background change on hover but rather a change in opacity for the icon/text itself.  I am suggesting a new prop `plain` to `v-btn` which will basically apply the following styles (as well as having the effect of passing the `flat` prop):

```
.btn--plain {
  height: auto;
  width: auto;
  margin: 0;
  padding: 8px;
  min-width: 0;
  > .btn__content {
    padding: 0;
    opacity: 0.75;
    &:before {
      background-color: transparent !important;
      transition: none !important;
    }
  }
  &:hover {
    > .btn__content {
      opacity: 1;
    }
  }
}
```
Here is an example of how it works/looks:
https://codepen.io/anon/pen/BJOYjd

I have been using this myself for a while and thought maybe others want something similar, so decided to suggest it as an addition to the `v-btn` component. What do you think? Is this something you will consider?

### Improvements
If this is functionality many users request it makes sense to include it in the framework.

### Bugs or Edge Cases it Helps Avoid
Users don't have to implement it themselves.

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
## Description

Resolves #3026 
## How Has This Been Tested?

Jest | Playground
## Markup:

<details>

```vue
<template>
  <v-col>
    <v-row align="center" justify="space-around">
      <v-btn :plain="isPlain" color="primary" @click="isPlain = !isPlain">plain btn</v-btn>
    </v-row>
  </v-col>
</template>

<script>
  export default {
    data () {
      return {
        isPlain: false,
      }
    },
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Updated Playground

<details>

```vue
<template>
  <div class="ma-12 pa-12">
    <v-row
      align="center"
      justify="space-around"
    >
      <v-btn depressed>
        Depressed
      </v-btn>
      <v-btn plain>
        Plain
      </v-btn>
      <v-btn text>
        Text
      </v-btn>
      <v-btn
        depressed
        color="primary"
      >
        Primary
      </v-btn>
      <v-btn
        depressed
        color="error"
      >
        Error
      </v-btn>
      <v-btn
        outlined
        color="success"
      >
        <v-icon left>mdi-vuetify</v-icon>

        Success
      </v-btn>
      <v-btn
        depressed
        disabled
      >
        Disabled
      </v-btn>
    </v-row>
    <div class="text-center">
      <v-btn
        class="ma-2"
        :loading="loading"
        :disabled="loading"
        color="secondary"
        @click="loader = 'loading'"
      >
        Accept Terms
      </v-btn>

      <v-btn
        :loading="loading3"
        :disabled="loading3"
        color="blue-grey"
        class="ma-2 white--text"
        @click="loader = 'loading3'"
      >
        Upload
        <v-icon
          right
          dark
        >
          mdi-cloud-upload
        </v-icon>
      </v-btn>

      <v-btn
        class="ma-2"
        :loading="loading2"
        :disabled="loading2"
        color="success"
        @click="loader = 'loading2'"
      >
        Custom Loader
        <template #loader>
          <span>Loading...</span>
        </template>
      </v-btn>

      <v-btn
        class="ma-2"
        :loading="loading4"
        :disabled="loading4"
        color="info"
        @click="loader = 'loading4'"
      >
        Icon Loader
        <template #loader>
          <span class="custom-loader">
            <v-icon light>mdi-cached</v-icon>
          </span>
        </template>
      </v-btn>

      <v-btn
        :loading="loading5"
        :disabled="loading5"
        color="blue-grey"
        class="ma-2 white--text"
        fab
        @click="loader = 'loading5'"
      >
        <v-icon dark>
          mdi-cloud-upload
        </v-icon>
      </v-btn>
    </div>
  </div>
</template>
<script>
  export default {
    data () {
      return {
        loader: null,
        loading: false,
        loading2: false,
        loading3: false,
        loading4: false,
        loading5: false,
      }
    },
    watch: {
      loader () {
        const l = this.loader
        this[l] = !this[l]

        setTimeout(() => (this[l] = false), 3000)

        this.loader = null
      },
    },
  }
</script>
<style>
  .custom-loader {
    animation: loader 1s infinite;
    display: flex;
  }
  @-moz-keyframes loader {
    from {
      transform: rotate(0);
    }
    to {
      transform: rotate(360deg);
    }
  }
  @-webkit-keyframes loader {
    from {
      transform: rotate(0);
    }
    to {
      transform: rotate(360deg);
    }
  }
  @-o-keyframes loader {
    from {
      transform: rotate(0);
    }
    to {
      transform: rotate(360deg);
    }
  }
  @keyframes loader {
    from {
      transform: rotate(0);
    }
    to {
      transform: rotate(360deg);
    }
  }
</style>
```

</details>

File: packages/vuetify/src/components/VBtn/VBtn.ts [hunk 1]
```diff
@@ -81,16 +82,19 @@ export default baseMixins.extend<options>().extend({
         'v-btn--absolute': this.absolute,
         'v-btn--block': this.block,
         'v-btn--bottom': this.bottom,
-        'v-btn--contained': this.contained,
-        'v-btn--depressed': (this.depressed) || this.outlined,
+        'v-btn--contained': this.isElevated, // TODO: remove v3
```
Comment (YipingRuan) [inline packages/vuetify/src/components/VBtn/VBtn.ts:85]: "isElevated" looks different from other names?  (should prefix is?)

Comment (BboyAkers) [reply packages/vuetify/src/components/VBtn/VBtn.ts:85]: It looks like it's a boolean. If the other properties are booleans should we adopt the 'is' syntax for future updates?

Comment (MajesticPotatoe) [reply packages/vuetify/src/components/VBtn/VBtn.ts:85]: things like `depressed` and `outlined` are public style props. `isElevated` is an internal computed prop and is in line with other computed props such as `isRound`

**Review by johnleider — APPROVED** [review]

**Pull request merged**